hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java

```diff
@@ -307,6 +312,61 @@ public void testAlreadyExistsException() {
         () -> catalog.createTable(TABLE_IDENTIFIER, schema, PartitionSpec.unpartitioned()));
   }
 
+  @Test
+  public void testCreateTableCommitFailure() throws TException, InterruptedException {
+    String namespace = DB_NAME;
+    String tableName = getRandomName();
+    TableIdentifier tableIdentifier = TableIdentifier.of(namespace, tableName);
+
+    HiveTableOperations hiveTableOperations =
+        (HiveTableOperations) catalog.newTableOps(tableIdentifier);
+
+    String tableLocation = catalog.defaultWarehouseLocation(tableIdentifier);
+    TableMetadata metadataV1 = createTableMetadata(tableLocation);
+
+    HiveTableOperations spyOps = spy(hiveTableOperations);
+    failCommitAndThrowException(spyOps, new InvalidObjectException());
+
+    AssertHelpers.assertThrows(
+        "Should throw invalid object exception",
+        ValidationException.class,
```

Comment:
it would be better to use `Assertions.assertThatThrownBy(...)` here, similar to https://github.com/apache/iceberg/blob/223177faf955bd8f11864477da16013cf5d7cc75/api/src/test/java/org/apache/iceberg/transforms/TestResiduals.java#L147-L149